assemblyline/common/net.py

```diff
@@ -21,6 +24,14 @@ def is_valid_port(value: int) -> bool:
     return False
 
 
+@functools.cache
+def find_top_level_domains():
+    """Combine (once and memoize) the three different sources of TLD."""
+    combined_tlds = TLDS_ALPHA_BY_DOMAIN.union({d for d in TLDS_SPECIAL_BY_DOMAIN if '.' not in d})
+    combined_tlds |= {tld.strip().upper() for tld in SYSTEM_LOCAL_TLD.split(";")}
```

Comment:
This will add an empty TLD if SYSTEM_LOCAL_TLD is not set.

```
>>> SYSTEM_LOCAL_TLD = os.getenv('SYSTEM_LOCAL_TLD', '')
>>> combined_tlds = {'a','b','c'} 
>>> combined_tlds |= {tld.strip().upper() for tld in SYSTEM_LOCAL_TLD.split(";")}
>>> combined_tlds
{'', 'a', 'b', 'c'}
```

It would probably be preferable if empty string would not end up in the TLDs

Reply:
how about now?